### PR TITLE
Being thrown, or leaping, over blood/oil decals do not dirty your feet/shoes anymore

### LINKED ISSUE #15532: Jumping Over Blood/Oil Puddles Counts as Stepping In It
title.

### PR DESCRIPTION
**Pull request by FluffyGhoster**
Being thrown, or leaping, over blood/oil decals do not dirty your feet/shoes anymore. Can still slip you.

Fixes #15532

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
!bugfix

**Comment by FluffyGhoster** [comment]
!review

**Review by alsoandanswer — APPROVED** [review]

**Review by SleepyGemmy — DISMISSED** [review]
being thrown or leaping over oil and blood shouldn't slip you.

**Comment by FluffyGhoster** [comment]
> being thrown or leaping over oil and blood shouldn't slip you.

Matter of opinion, people don't perfectly jump as far as they intended, which is why it checks the slip probability even if you're jumping over it

**Comment by SleepyGemmy** [comment]
are your shoes dirtied if this slip probability occurs?

**Review by Alberyk — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by SleepyGemmy** [comment]
> > are your shoes dirtied if this slip probability occurs?
> 
> No, you slip onto the landing area, not on the turf itself

fair enough.

**Review by SleepyGemmy — APPROVED** [review]

File: code/game/objects/effects/decals/Cleanable/humans.dm [hunk 1]
```diff
@@ -89,41 +89,50 @@
...
+
+		else if (hasfeet)//Or feet
+			perp.footprint_color = basecolor
+			perp.track_footprint = max(amount,perp.track_footprint)
+			LAZYINITLIST(perp.feet_blood_DNA)
+			if (blood_DNA)
+				perp.feet_blood_DNA |= blood_DNA.Copy()
+		else if (perp.buckled_to && istype(perp.buckled_to, /obj/structure/bed/stool/chair/office/wheelchair))
+			var/obj/structure/bed/stool/chair/office/wheelchair/W = perp.buckled_to
+			W.bloodiness = 4
```
Comment (SleepyGemmy) [inline code/game/objects/effects/decals/Cleanable/humans.dm:124]:
```suggestion
		else if(hasfeet) //Or feet
			perp.footprint_color = basecolor
			perp.track_footprint = max(amount,perp.track_footprint)
			LAZYINITLIST(perp.feet_blood_DNA)
			if(blood_DNA)
				perp.feet_blood_DNA |= blood_DNA.Copy()
		else if(perp.buckled_to && istype(perp.buckled_to, /obj/structure/bed/stool/chair/office/wheelchair))
			var/obj/structure/bed/stool/chair/office/wheelchair/W = perp.buckled_to
			W.bloodiness = 4
```

**Pull request merged**